app/containers/container_FBLogin.js

```diff
@@ -0,0 +1,108 @@
+import React, {
+  Component,
+  StyleSheet,
+  Image,
+  Text,
+  View,
+} from 'react-native';
+import { connect } from 'react-redux';
+import { bindActionCreators } from 'redux';
+import * as userActions from '../actions/action_user';
+import { Actions } from 'react-native-router-flux';
+import { changeUser } from '../lib/db/db';
+
+
+
+import FBLogin from 'react-native-facebook-login';
+
+//sets fb photo width
+const FB_PHOTO_WIDTH = 200;
+
+//this is data.credentials
+// { tokenExpirationDate: '2016-05-22T17:41:07-07:00',
+//   permissions: [ 'email', 'contact_email', 'user_friends', 'public_profile' ],
+//   userId: '10153780375328598',
+//   token: 'CAAYZANvux7B...etc..gBAOCT0knZg' }
+
+class LogIn extends Component {
+  constructor(props){
+    super(props);
+  }
+
+  render() {
+  const { user, action } = this.props;
+    var _this = this;
+    return (
+      <View style={styles.loginContainer}>
+
+        <FBLogin style={{ marginBottom: 10, }}
+          permissions={["email","user_friends"]}
+          onLogin={function(data){
+            console.log("*******Logged in! and this is props, this is data cred", data.credentials);
+            changeUser(data.credentials.userId)
+            Actions.view();
+            action.firebase_check(data.credentials);
+          }}
+          onLogout={function(){
+            console.log("*******Logged out.");
+            action.logOut();
+          }}
+          onLoginFound={function(data){
+            //if user was already signed in, no need to sign in again
+            console.log("********Existing login found.");
+            changeUser(data.credentials.userId)
+            Actions.view();
+            action.firebase_check(data.credentials);
+          }}
+          onLoginNotFound={function(){
+            //if user has not signed in since last session
+            console.log("*******No user logged in.");
+            action.logOut();
+          }}
+          onError={function(data){
+            console.log("*******ERROR");
+            console.log(data);
+          }}
+          onCancel={function(){
+            console.log("******User cancelled.");
+          }}
+          onPermissionsMissing={function(data){
+            console.log("*******Check permissions!");
+            console.log(data);
+          }}
+        />
+
+        <Text>{ user ? user.token : "N/A" }</Text>
```

Comment:
@jake-shasteen this was for my own testing before I changed state--> redux state. I'll take this out!